base58/base58.go

```diff
@@ -21,8 +19,8 @@ func Decode(b string) []byte {
 	j := big.NewInt(1)
 
 	for i := len(b) - 1; i >= 0; i-- {
-		tmp := strings.IndexAny(alphabet, string(b[i]))
-		if tmp == -1 {
+		tmp := b58[b[i]]
```

Comment:
Disregard this.  I updated my comment above about the scratch with a way to avoid the new big.Int there too.